### PR TITLE
Tomw/player message fixes

### PR DESCRIPTION
**Pull request by Alphy11**

### CODE REVIEW

File: client/src/PlayerLanding/index.tsx [hunk 1]
```diff
@@ -47,8 +47,10 @@ function PlayerLanding() {
       </>
     );
   }
-
-  if (Object.keys(game.roleBag ?? {}).length === 0) {
```
Comment (Alphy11) [inline client/src/PlayerLanding/index.tsx:51]: @vermentor why was this here?  This file is by far the easiest one to break things, so 🤷‍♂️ but I am not sure why we'd be in a situation where the role bag is empty, and a player is here.  And it is preventing players from rejoining a game when they have a role.  In fact, I am not sure how it's not completely broken

Comment (Alphy11) [reply client/src/PlayerLanding/index.tsx:51]: My guess is that when you added acknowledge role, you didn't add it to test game creator and so maybe this doesn't get hit in real life?

Comment (vermentor) [reply client/src/PlayerLanding/index.tsx:51]: Hmm this was before test game creator.  Role bag is empty until the moment the storyteller starts the game and roles become available to select for players, so that's probably why this existed.  During this time the alternative paths are PlayerRoleSelect (invalid), PlayerRole (invalid), and PlayerInGame (also invalid).

I'd be careful when testing that when a player joins it doesn't take them directly to the PlayerRoleSelect page while seating is still settling.  Although that might actually be fine so long as it brings them back in case of ordering problems.

File: client/src/GamemasterDesktop/DesktopView.tsx [hunk 2]
```diff
@@ -72,9 +92,11 @@ function Grimoire({ isPlayerView = true }: GrimoireProps) {
                 <TeamDistributionBar />
                 <div className="text-center">
                   <div>Alive players: {alivePlayers.length}</div>
-                  <div>{`Votes to execute: ${Math.ceil(
-                    alivePlayers.length / 2,
-                  )}`}</div>
+                  <div className="capitalize">{`Currently executeing: ${
+                    playerOnBlock.player ?? "Nobody"
+                  }`}</div>
+                  <div>{`Votes needed : ${playerOnBlock.votes}`}</div>
```
Comment (vermentor) [inline client/src/GamemasterDesktop/DesktopView.tsx:98]: I find "Votes needed" just slightly confusing, since it can be - but isn't necessarily - votes to tie.

**Review by vermentor — APPROVED** [review]
